## **1. Data Acquisition**

#### **Step 1. Using Frasa**

Untuk melakukan crawling lirik lagu, kami menambahkan modul crawling pada library **Frasa (0.1.58)**.

In [1]:
# %pip install frasa==0.1.58

#### **Step 2. Import & Define Artist**

Import library dan siapkan nama artis yang ingin di unduh lirik-lirik lagunya.

In [11]:
from slugify import slugify

artist_name = 'Avenged Sevenfold'
artist_slug = slugify(artist_name)

#### **Step 3. Scraping List of Song**

Lakukan penarikan data menggunakan pustaka **frasa** sesuai dengan nama artis yang sudah di definisikan sebelumnya.

In [12]:
from frasa.azlyrics import azlyrics
import pandas as pd

api = azlyrics()
api.artist = artist_name
lists = api.lagu()
 
df = pd.DataFrame.from_dict(lists, orient='index')
df.index.name = 'Judul'

print("Total Lagu: " + str(len(df)))

Total Lagu: 116


Dari proses diatas, ditemukan sebanyak 116 lirik lagu dari Avenged Sevenfold yang di deteksi. Berikut ini tabelnya:

In [14]:
df.head()

year                         album   type  \
Judul                                                                         
To End The Rapture                2001  Sounding The Seventh Trumpet  album   
Turn The Other Way                2001  Sounding The Seventh Trumpet  album   
Darkness Surrounding              2001  Sounding The Seventh Trumpet  album   
The Art Of Subconscious Illusion  2001  Sounding The Seventh Trumpet  album   
We Come Out At Night              2001  Sounding The Seventh Trumpet  album   

                                                                                url  
Judul                                                                                
To End The Rapture                http://www.azlyrics.com/lyrics/avengedsevenfol...  
Turn The Other Way                http://www.azlyrics.com/lyrics/avengedsevenfol...  
Darkness Surrounding              http://www.azlyrics.com/lyrics/avengedsevenfol...  
The Art Of Subconscious Illusion  http://www.azlyrics.com/lyrics/avengedsevenfol...  
We Come Out At Night              http://www.azlyrics.com/lyrics/avengedsevenfol...

Selanjutnya urutkan data berdasaarkan `Judul`-nya dan simpan sebagai file `csv` di direktori `unduhan`.

In [15]:
df = df.sort_values('Judul')
df.to_csv(f'unduhan/{artist_slug}.csv')

#### **Step 4. Scraping Lyrics**

Terakhir, unduh semua lirik dari seluruh daftar lagu yang terdeteksi menggunakan **frasa**.

In [16]:
import os 

judul_data = df.index.tolist()

for judul in judul_data:
    slug = slugify(judul)
    
    lirik = azlyrics()
    lirik.artist = artist_name
    lirik.title = judul
    
    file_path = f"unduhan/{artist_slug}/{slug}.txt"
    if os.path.exists(file_path):
        print("Lirik sudah ada. Lanjut.")
    else:
        with open(file_path, 'w') as txtfile:
            txtfile.write(str(lirik.lirik()))

print('Lirik berhasil di unduh.')

Lirik berhasil di unduh.


Setelah proses ini dilakukan, kami memiliki direktori file baru berikut:

- ``unduhan/avenged-sevenfold.csv``
- ``unduhan/avenged-sevenfold/*.txt``